vassal-app/src/main/java/VASSAL/counters/BasicPiece.java

```diff
@@ -889,7 +891,7 @@ else if (getMap() != null && stroke.equals(getMap().getStackMetrics().getMoveBot
   @Override
   public String getState() {
     final SequenceEncoder se = new SequenceEncoder(';');
-    final String mapName = map == null ? "null" : map.getIdentifier(); // NON-NLS
+    final String mapName = getMap() == null ? "null" : getMap().getIdentifier(); // NON-NLS
```

Comment:
Should getMap() just get called once and stored?

Reply:
I don't think so as map can change over time between calls to getState().

Reply:
I mean in that one line you’re calling it twice (null check & then using it). Sort of like calling getGameModule over and over?

Reply:
I've adjusted this:
```
    final Map m = getMap();
    final String mapName = m == null ? "null" : m.getIdentifier(); // NON-NLS
```